Accumulated changes to update to current OS and software versions

wtoorop opened the pull request:
I'm keeping this draft for now because more is coming...

paulehoffman commented:
Excellent stuff! Keep them coming.

paulehoffman commented:
Looks good, thanks!

As an aside, I had never seen "not a in b" written as "a not in b" and had to go look at the language reference, and there it is.

gthess committed:
Bare minimum for Vagrant provisioning

gthess committed:
Accept both new and old Ansible inventory variables

gthess committed:
Better variable default and remove TIME_WAIT for BSD VMs

wtoorop committed:
leftover fix

wtoorop committed:
Try to setup bridge_net on FreeBSD only

As workaround of trying to setup things which are specific to our shared machine.
We should probably do this differently

gthess committed:
Finalising the provision procedure

gthess committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
Always run the ansible provisioner to keep the inventory up to date

gthess committed:
Install packeges needed for resolver compilation

gthess committed:
Remove non existing cli command

gthess committed:
Fix knot gpg key

wtoorop committed:
Sketch extending host_vars file

wtoorop committed:
Note about our infra specific network interfaces

gthess committed:
Explicit v4 static routes for resolvers<>gateway<>servers

gthess committed:
Still trying static routes in linux

gthess committed:
Fix unused variable

gthess committed:
Finalise static route changes

gthess committed:
wip for parallel compilation of resolvers

gthess committed:
Update Vagrant file

gthess committed:
Added local configuration logic and some documentation

wtoorop committed:
Configure vm interfaces other than BRIDGE_NET too

wtoorop committed:
Correct one unused resolvers-vm IP

gthess committed:
Added VirtualBox linked clones as a local configuration option; default
off for now.

gthess committed:
Merge branch 'master' into yorgos_wip

gthess committed:
Fetch and decompress all software sources

gthess committed:
Explicitly specify the owner:group for unarchived files

gthess committed:
Add resolver compilation procedure

gthess committed:
Better templating and complilation done marking

gthess committed:
Use 'free' strategy for playbook readability

gthess committed:
Add patches for pdns

gthess committed:
Handle None iteration

gthess committed:
Fix the ansible conditional

Commit:
Make custom IPv4 addresses for the root servers consistent

Changed the last 4 addresses from 172.21.109.1[0123] to 172.21.11[0123].1, similar to the way the first 9 are numbered

Commit:
Add root zone setup for scenario 2 (section 5.2 in RSSAC028)

This setup can be provisioned on servers-vm with the ansible playbook ansible/scenario-2_network_provision_servers-vm.yml

gthess committed:
Do not use 'strategy: free' as the output is hard to grasp

gthess committed:
Update doc

gthess committed:
More jobs for make

gthess committed:
Merge pull request #1 from NLnetLabs/yorgos_wip

Compile resolvers in parallel

gthess committed:
Mention vagrant reloading in the README

gthess committed:
Merge pull request #2 from NLnetLabs/jelte-fix-local-root-addresses

Make custom IPv4 addresses for the root servers consistent

gthess committed:
Use 'all' group variables for building directories

gthess committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
- Programmatically create the index for VBoxManage commands and use an explicit
  NIC type.

gthess committed:
Don't change the current default FreeBSD box yet.

gthess committed:
More generalisation for FreeBSD networking and building software

gthess committed:
Change the FreeBSD image

gthess committed:
Update template

gthess committed:
Better naming for skipped tasks

gthess committed:
Configuring resolv.conf is part of the network_provision.yml now

gthess committed:
Testing connectivity

gthess committed:
Clean up

gthess committed:
Rename test

gthess committed:
Changed FreeBSD box

gthess committed:
Change the default IP scheme from 172.2x.1.x to 172.2x.0.x

gthess committed:
Template away named.conf

gthess committed:
Typo

gthess committed:
Typo

Commit:
Add root zone setup for scenario 3 (section 5.3 in RSSAC028)

This setup can be provisioned on servers-vm with the ansible playbook ansible/scenario-3_network_provision_servers-vm.yml

In this scenario, the root server addresses are added as direct A and AAAA records in the root zone, and signed as such. There is no delegation to a 'root server zone'. The particular setup chosen here is to also not use an ENT, i.e. the a root server's FQDN is 'a.', etc.

Commit:
Add root zone setup for scenario 4 (section 5.4 in RSSAC028)

This setup can be provisioned on servers-vm with the ansible playbook ansible/scenario-4_network_provision_servers-vm.yml

In this scenario, there is no delegation through an externally hosted zone. The root servers have their addresses in a TLD specifically for this purpose.

In this particular setup, the TLD is 'some-servers', with the FQDN of the first root server being 'a.some-servers.'.

gthess committed:
Network provision different OSes

moritzcm committed:
Added PDNS patches

patched pdsn

compile pdns

tjeb committed:
Add root zone setup for scenario 5 (section 5.5 in RSSAC028)

This setup can be provisioned on servers-vm with the ansible playbook ansible/scenario-5_network_provision_servers-vm.yml

In this particular setup, the added TLDs are 'a.' up to 'm.', signed and delegated from the root zone. The nameserver addresses within these zones are 'ns.<zone>. A' and 'ns.<zone>. AAAA'.

tjeb committed:
Add root zone setup for scenario 6 (section 5.6 in RSSAC028)

This setup can be provisioned on servers-vm with the ansible playbook ansible/scenario-6_network_provision_servers-vm.yml

In this particular setup, the entire NS set for the root is:

. $TTL IN NS all-some-servers.

With 'all-some-servers.' having a single A and AAAA, each with the 13 root server addresses, directly in the root zone.

gthess committed:
Do not disable NetworkManager in CentOS

gthess committed:
Use eth0 for the first (Vagrant) interface; more compatibility with
older OS versions.

gthess committed:
Disable synced folder

gthess committed:
Some more documentation

gthess committed:
Add playbook for faster environment setup

gthess committed:
Merge pull request #4 from NLnetLabs/yorgos_wip

More generalisation for network configurations and IP scheme change for the base testbed VMs and faster vagrant/virtualbox environment setup

gthess committed:
Document ansible-playbook lower fork number for machines with low core count

wtoorop committed:
tcpdumpd service for gateway

gthess committed:
Merge pull request #5 from NLnetLabs/moritz_wip

Added PDNS patches

wtoorop committed:
Some ground work for running tests

wtoorop committed:
Merge pull request #6 from NLnetLabs/jelte-add-root-scenarios

Jelte add root scenarios

wtoorop committed:
Zones and nameserver configuration for rerunning appendix A of rssac028

wtoorop committed:
Pretty printing of the RSSAC028 Appendix A results

wtoorop committed:
Merge pull request #7 from NLnetLabs/rerun-rssac028-appendix-a

Rerun rssac028 appendix a

gthess committed:
Introduce inception for parallel compilation on multiple hosts

gthess committed:
Use correct file

gthess committed:
Try with python ansible on old ubuntu

gthess committed:
Use default pip

gthess committed:
Prefer Ansible from pip

gthess committed:
Per host packages and same Ansible version for inception

gthess committed:
Use proper namespace for tasks now that the ansible version for inception is pinned

gthess committed:
Remove any leftover inception state

gthess committed:
Try generic packages for centos

wtoorop committed:
Bugfix in nsd conf and building old version from ports

wtoorop committed:
And actually run the scenarios

wtoorop committed:
Generate more realistic zones

wtoorop committed:
Generated zone files with suggested names and current IPs

gthess committed:
Fix identation

gthess committed:
Ensure latest pip

gthess committed:
Add /usr/local/bin to PATH in CentOS

wtoorop committed:
.net not served from the root-servers

wtoorop committed:
Facilitate testing non-existing root-servers.net

gthess committed:
Compile on CentOS systems

gthess committed:
Also AlmaLinux

gthess committed:
Remove unused task

gthess committed:
Merge pull request #8 from NLnetLabs/yorgos_wip

Compiling software on VMs in parallel

gthess committed:
Trying metric on routes for centos/alma

gthess committed:
Prefer system ansible where it works, fixes for CentOS, NetBSD, and pinf connectivity test

gthess committed:
Merge pull request #9 from NLnetLabs/yorgos_wip

Fixes for CentOS, NetBSD and ping connectivity test

wtoorop committed:
Make bind-10.3 build work

wtoorop committed:
Jelte's loadbalancing work

wtoorop committed:
Willem's tcpdump work

wtoorop committed:
Capture all interfaces

gthess committed:
- gateway-vm does IPv4 NATing for other VMs using it as a default route;
- better configuration and templates for multiple routes per interface
  (mainly for adding default routing);
- no restart required on network provisioning.

gthess committed:
Forgot handlers on root servers playbook

gthess committed:
Fix template

gthess committed:
Fix netbsd template and network restart; former also for Ubuntu

gthess committed:
Also restart static routes for netbsd

gthess committed:
Apply netplan in Ubuntu

gthess committed:
Fixes for NetBSD network restart, resolvconf, centos route6 template

gthess committed:
Fix default variables for ansible

gthess committed:
- Neater network provision output.
- Fix possible vagrant locking while provisioning in parallel.

gthess committed:
Update documentation

wtoorop committed:
Do Appendix A queries with dig

gthess committed:
Prepare for nameserver run

gthess committed:
Install rsync on servers-vm

gthess committed:
Reorder compilation step

gthess committed:
Network restart more idempotent

gthess committed:
Remove leftover handler notify

gthess committed:
Moved inventory to a different level than playbooks to allow for better
controlled variable preference when mixing different inventories.

gthess committed:
Merge branch 'yorgos_wip'

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Point ansible to new hosts location

gthess committed:
Build software to list instead of dictionary; will help with assigning IPs later

gthess committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Provision loadbalancer with actual root-server IPs

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
Retry when bringing up the machine; avoids rare slowness between Vagrant and VirtualBox

gthess committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
Properly set the retry in the playbook

gthess committed:
More fine grained system information for the different specific tasks

gthess committed:
wip for configuring/starting/testing nameservers

gthess committed:
Fix build configuration. it is a list now

gthess committed:
Fix identation

gthess committed:
MSS will be configured on the interface

wtoorop committed:
Merge branch 'master' into running-tests

wtoorop committed:
Adapt running-tests work to current master

gthess committed:
Declutter output when installing packages on different systems

gthess committed:
There are no handlers anymore

gthess committed:
Simpler import and proper failure for system specific tasks

gthess committed:
For generic packages we need an intermediate steps to set default values

gthess committed:
If scenario is specified, configure the relevant zones

gthess committed:
Disable SELinux on CentOS

gthess committed:
Merge branch 'master' into yorgos_wip

gthess committed:
Merge branch 'master' into yorgos_wip

gthess committed:
Add rc.d service files

gthess committed:
Fix leftover variable

gthess committed:
Leftover variables

gthess committed:
pidfile for rc.d and bind

gthess committed:
More restrictions for the service name

wtoorop committed:
Run resolver tests

gthess committed:
Fix linking for bind and libuv on netbsd

gthess committed:
Quote the pid filename in bind

gthess committed:
Make bind work with centos systemd

gthess committed:
Remove any firewall on root servers; centos was by default on

gthess committed:
Test nameservers (dig) from the resolvers-vm

gthess committed:
Clean gateway provisioning

gthess committed:
Fix task names

gthess committed:
Merge branch 'master' into yorgos_wip

gthess committed:
Add hostname to nameserver version

gthess committed:
- Explicitly set LC_ALL on the environment for incepted Ansible;
- Cleaner file synchroniastion for incepted Asnsible;
- Let the dig test to wait a little bit longer.

gthess committed:
Trying without resolv.conf changes

wtoorop committed:
Playbook to configure individual scenario/software combinations on servers-vm

gthess committed:
Test for the MTU settings

gthess committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
Trying MTU with rc.local

wtoorop committed:
Update selection of scenarios for all variants of scenarios

Including variants for scenarios 3, 5 and 6
Including using the real ips
Including a single letter deployment for scenario 5

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

gthess committed:
Optional box_version to pass to Vagrant

gthess committed:
Testing MTU with rc.local for CentOS

gthess committed:
For netbsd always reply 'yes' to fsck for reboots; maybe disabling in the future is better

gthess committed:
Trying rc.local with systemd

gthess committed:
Script executable for systemd

gthess committed:
Use absolute path to please all relevant OSes

gthess committed:
Trying MTU with plain rc.local on FreeBSD

gthess committed:
Better templating

gthess committed:
Accurate service restart

gthess committed:
Accurater service restart

gthess committed:
Trying MSS for Linux

gthess committed:
Typo in template

gthess committed:
More typos in template

gthess committed:
Proper ROUTE configuration

gthess committed:
Also for v6

gthess committed:
Fix servers-vm service configuration

wtoorop committed:
Correctly select scenario on root-servers

- Including configuring servers-vm for .net when needed
- Using real IPs bny default
- With variants

gthess committed:
More robust net_ping

wtoorop committed:
Make sure .net is available

gthess committed:
Merge branch 'master' into yorgos_wip

gthess committed:
Accomodate net_ping for servers-vm artificial delay

gthess committed:
Fix template and comment on bug

wtoorop committed:
Provision load_balancer for individual IPs

wtoorop committed:
list_root-servers.yml creates results ips2root_server_software.csv

wtoorop committed:
Work around multiple ips for one software issue with aj-linux

wtoorop committed:
Start bind resolvers with service files

wtoorop committed:
All resolver starting/stopping service files

wtoorop committed:
Also use the PowerDNS resolvers

wtoorop committed:
Root servers listen on adresses, not networks

wtoorop committed:
Provide filename for pcap to save

wtoorop committed:
Normalize real IPv6 addresses of root servers

wtoorop committed:
ipinfo.json instead of the root-servers IP list

wtoorop committed:
Bugfix ipinfo.json to contain actual json

wtoorop committed:
Update ipinfo

gthess committed:
Update templates for cookie configuration

gthess committed:
Fix bind service template

gthess committed:
Fix bind service template again

wtoorop committed:
Special to query domain for servers-vm

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Special example. domain for test query

Also nsd-4.1.20 does not know about cookie-secret

wtoorop committed:
Also capture historic (and future) root IP addresses

wtoorop committed:
Second testing tld: example2.

wtoorop committed:
Send a second test query.

wtoorop committed:
regenerating root zones for example2. delegation

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Signed versions of example and example2 zones

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Capture names start with date

wtoorop committed:
Generate trust anchors from the source and fix trust anchor for bind >= 9.15.8

wtoorop committed:
Make resolvers less noisy

wtoorop committed:
typo

gthess committed:
Add more resolver software

gthess committed:
Adjust template configuration

gthess committed:
Fix conflict and ready2run knot-resolver

gthess committed:
Add luajit for knot-resolver

gthess committed:
Merge pull request #10 from NLnetLabs/more-resolvers

More resolvers

gthess committed:
Fix pdns service file

wtoorop committed:
Fix service and configuration for newer pdns recursors

wtoorop committed:
Provision and configure Cloudflare authoritative simulation with RSSAC028 version of ldns-testns

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
Empty string with pdns is no quotes either

wtoorop committed:
Simulate Verisign Atlas with ldns-testns

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

wtoorop committed:
No more exceptions for aj-linux

wtoorop committed:
Scenarios with fake IPs

wtoorop committed:
Handle realips setting for .net on servers-vm

wtoorop committed:
Better comment in the template

wtoorop committed:
Delete knot-resolver persistent cache before querying

wtoorop committed:
Root hints for the different scenarios

wtoorop committed:
Merge branch 'master' of github.com:NLnetLabs/resolver-testbed

Commit:
Resolve with root.hints

wtoorop committed:
Base trust anchor on the generated root KSK (by constant name)

wtoorop committed:
Fix creating zones for BSD's

wtoorop committed:
Had to move the LD_LIBRARY_PATH form stop command to work

wtoorop committed:
Not logging from ldns-testns makes them more responsive

wtoorop committed:
Swap 5.* with 9.*

wtoorop committed:
Hide real IP zone files and alternative IP hints files

wtoorop committed:
Swap deployment of realips zonefiles

wtoorop committed:
Remove explicit references to certain proprietary software
